sdk/core/azure-core/inc/azure/core/http/curl/curl_session.hpp

```diff
@@ -342,7 +353,7 @@ namespace Azure { namespace Core { namespace Http {
       // By not moving the connection back to the pool, it gets destroyed calling the connection
       // destructor to clean libcurl handle and close the connection.
       // IsEOF will also handle a connection that fail to complete an upload request.
-      if (IsEOF())
+      if (IsEOF() && m_keepAlive)
```

Comment:
Is this covered by tests? Is this the only place we need to account for keep alive?

Reply:
yes, there is a test for this.
And yes, this is the only place where we decide if we move a connection back to the connection pool or not